File: pkg/dev_compiler/test/codegen_expected/language/number_identity2_test.js
dart_library.library('language/number_identity2_test', null, /* Imports */[
  'dart_sdk',
  'expect'
], function load__number_identity2_test(exports, dart_sdk, expect) {
  'use strict';
  const core = dart_sdk.core;
  const typed_data = dart_sdk.typed_data;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const expect$ = expect.expect;
  const number_identity2_test = Object.create(null);
  let intTodouble = () => (intTodouble = dart.constFn(dart.definiteFunctionType(core.double, [core.int])))();
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  number_identity2_test.uint64toDouble = function(i) {
    let buffer = typed_data.Uint8List.new(8)[dartx.buffer];
    let bdata = typed_data.ByteData.view(buffer);
    bdata[dartx.setUint64](0, i);
    return bdata[dartx.getFloat64](0);
  };
  dart.fn(number_identity2_test.uint64toDouble, intTodouble());
  number_identity2_test.testNumberIdentity = function() {
    let a = core.double.NAN;
    let b = dart.notNull(a) + 0.0;
    expect$.Expect.isTrue(core.identical(a, b));
    a = number_identity2_test.uint64toDouble((1)[dartx['<<']](64) - 1);
    b = number_identity2_test.uint64toDouble((1)[dartx['<<']](64) - 2);
    expect$.Expect.isFalse(core.identical(a, b));
    a = 0.0 / 0.0;
    b = 1.0 / 0.0;
    expect$.Expect.isFalse(core.identical(a, b));
  };
  dart.fn(number_identity2_test.testNumberIdentity, VoidTodynamic());
  number_identity2_test.main = function() {
    for (let i = 0; i < 20; i++) {
      number_identity2_test.testNumberIdentity();
    }
  };
  dart.fn(number_identity2_test.main, VoidTodynamic());
  // Exports:
  exports.number_identity2_test = number_identity2_test;
});

File: tests/language_2/number_identity2_test.dart
// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.
// Dart test program for testing params.
//
// Contains test that is failing on dart2js. Merge this test with
// 'number_identity_test.dart' once fixed.
// VMOptions=--optimization-counter-threshold=10

import "package:expect/expect.dart";
import 'dart:typed_data';

double uint64toDouble(int i) {
  var buffer = new Uint8List(8).buffer;
  var bdata = new ByteData.view(buffer);
  bdata.setUint64(0, i);
  return bdata.getFloat64(0);
}

testNumberIdentity() {
  var a = double.NAN;
  var b = a + 0.0;
  Expect.isTrue(identical(a, b));

  a = uint64toDouble((1 << 64) - 1);
  b = uint64toDouble((1 << 64) - 2);
  Expect.isFalse(identical(a, b));

  a = 0.0 / 0.0;
  b = 1.0 / 0.0;
  Expect.isFalse(identical(a, b));
}

main() {
  for (int i = 0; i < 20; i++) {
    testNumberIdentity();
  }
}
